File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file265.f90
subroutine kernel_jacobi_2d_imper(tsteps, n, a, b)
implicit none

DATA_TYPE, dimension(n, n) :: a
DATA_TYPE, dimension(n, n) :: b
integer :: n, tsteps
integer :: i, j, t

!$pragma scop
do t = 1, _PB_TSTEPS
do i = 2, _PB_N - 1
do j = 2, _PB_N - 1
b(j, i) = 0.2D0 * (a(j, i) + a(j - 1, i) + a(1 + j, i) + &
a(j, 1 + i) + a(j, i - 1))
end do
end do
do i = 2, _PB_N - 1
do j = 2, _PB_N - 1
a(j, i) = b(j, i)
end do
end do
end do
!$pragma endscop
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file265.cpp
#include <stdio.h>

#define DATA_TYPE double
#define _PB_N 100
#define _PB_TSTEPS 100

DATA_TYPE a[_PB_N][_PB_N];
DATA_TYPE b[_PB_N][_PB_N];

void kernel_jacobi_2d_imper(int tsteps, int n, DATA_TYPE a[_PB_N][_PB_N], DATA_TYPE b[_PB_N][_PB_N]) {
  int i, j, t;
  for (t = 1; t < _PB_TSTEPS; t++) {
    for (i = 2; i < _PB_N - 1; i++) {
      for (j = 2; j < _PB_N - 1; j++) {
        b[j][i] = 0.2 * (a[j][i] + a[j-1][i] + a[j+1][i] + a[j][i+1] + a[j][i-1]);
      }
    }
    for (i = 2; i < _PB_N - 1; i++) {
      for (j = 2; j < _PB_N - 1; j++) {
        a[j][i] = b[j][i];
      }
    }
  }
}
